### PR TITLE
223-text-prediction-updated

### PR DESCRIPTION
**Pull request by dwipddalal**
223-text-prediction-updated

### CODE REVIEW

**Comment by dwipddalal** [comment]
Is this PR good or are there any changes that I need to make in this case?

File: notebooks/223-text-prediction/README.md
Comment (adrianboguszewski) [inline notebooks/223-text-prediction/README.md:3]:
```suggestion
In this demo we have used **Generative Pre-trained Transformer 2 ([GPT-2](https://github.com/openvinotoolkit/open_model_zoo/tree/master/models/public/gpt-2))** model for text prediction.
```

File: notebooks/223-text-prediction/README.md
Comment (adrianboguszewski) [inline notebooks/223-text-prediction/README.md:13]:
```suggestion
This is an interactive demonstration in which the user can type text into the input bar and generate predicted text. This procedure can be repeated as many times as the user desires.

```

File: notebooks/223-text-prediction/README.md
Comment (adrianboguszewski) [inline notebooks/223-text-prediction/README.md:19]:
```suggestion
```

File: notebooks/README.md
Comment (adrianboguszewski) [inline notebooks/README.md:146]:
```suggestion
| [223-text-prediction](223-text-prediction/)<br> | Use Gpt-2 to perform text prediction on the input sequence | <img src=https://user-images.githubusercontent.com/91228207/185105225-0f996b0b-0a3b-4486-872d-364ac6fab68b.png  width=225> |
```

File: README.md
Comment (adrianboguszewski) [inline README.md:146]:
```suggestion
| [223-text-prediction](notebooks/223-text-prediction/)<br> | Use Gpt-2 to perform text prediction on the input sequence | <img src=https://user-images.githubusercontent.com/91228207/185105225-0f996b0b-0a3b-4486-872d-364ac6fab68b.png  width=225> |
```

**Review by adrianboguszewski — CHANGES_REQUESTED** [review]
Great work @dwipddalal! We need just minor changes. Please commit my suggestions as well as fix the notebook for CI to pass. All tests must be green to merge your PR. Run nbqa on your local machine to fix the issues below:
Screenshot from 2022-08-18 14-35-17

**Comment by dwipddalal** [comment]
> Great work @dwipddalal! We need just minor changes. Please commit my suggestions as well as fix the notebook for CI to pass. All tests must be green to merge your PR. Run nbqa on your local machine to check the issues below: Screenshot from 2022-08-18 14-35-17 Finally, clear your notebooks outputs by running: `jupyter nbconvert --clear-output --inplace 223-text-prediction.ipynb`

Sure. I shall resolve them as soon as possible.

**Review by adrianboguszewski — APPROVED** [review]

File: notebooks/223-text-prediction/223-text-prediction.ipynb
Comment (adrianboguszewski) [inline notebooks/223-text-prediction/223-text-prediction.ipynb:420]:
```suggestion
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
  },
```

File: notebooks/223-text-prediction/223-text-prediction.ipynb
Comment (adrianboguszewski) [inline notebooks/223-text-prediction/223-text-prediction.ipynb:422]:
```suggestion
```

File: notebooks/223-text-prediction/223-text-prediction.ipynb
Comment (adrianboguszewski) [inline notebooks/223-text-prediction/223-text-prediction.ipynb:420]:
```suggestion
  },
```

**Comment by dwipddalal** [comment]
We need to pass these 18 test cases, right? If so, then I shall read about them in more detail and try to find the error.

**Comment by adrianboguszewski** [comment]
Yes. We must find the cause of these errors and fix it.

File: notebooks/223-text-prediction/223-text-prediction.ipynb
Comment (adrianboguszewski) [inline notebooks/223-text-prediction/223-text-prediction.ipynb:125]:
```suggestion
    "        f\" --download_dir {base_model_dir} --output_dir {base_model_dir}\"\n",
```

**Comment by adrianboguszewski** [comment]
@dwipddalal I found and fixed the issue. Other errors are related to external packages and will be fixed in another PR.

**Comment by dwipddalal** [comment]
It's great to hear that. I shall try to remove the other errors.

**Comment by adrianboguszewski** [comment]
Thanks for these changes. We have to merge #619 before this to fix CI issues.

**Comment by dwipddalal** [comment]
So here, first, we shall merge the pr and then focus on fixing the remaining issues? For merging Ig, we shall first need approval from one more reviewer.

**Comment by dwipddalal** [comment]
Okay.

**Comment by dwipddalal** [comment]
If the issue of external packages is solved, then shall I once again push the previous commit?

**Comment by adrianboguszewski** [comment]
You don't have to do anything related to external packages. But let's wait for the @ryanloney review.

**Review by ryanloney — CHANGES_REQUESTED** [review]
@dmatveev please review suggested changes and update the name of the directory and .ipynb to include "gpt2" 

223-gpt2-text-prediction
223-gpt2-text-prediction.ipynb

**Comment by dwipddalal** [comment]
> @dwipddalal please review suggested changes and update the name of the directory and .ipynb to include "gpt2"
> 
> 223-gpt2-text-prediction 223-gpt2-text-prediction.ipynb

Changed both the file name as well as the folder name.

File: notebooks/223-text-prediction/README.md
Comment (adrianboguszewski) [inline notebooks/223-text-prediction/README.md:9]:
```suggestion
This is a demonstration in which the user can type the beginning of the text and the network will generate a further. This procedure can be repeated as many times as the user desires.
```

File: notebooks/223-gpt2-text-prediction/223-gpt2-text-prediction.ipynb
Comment (adrianboguszewski) [inline notebooks/223-gpt2-text-prediction/223-gpt2-text-prediction.ipynb:13]:
```suggestion
    "This notebook shows a text prediction with OpenVINO. We use the [GPT-2](https://github.com/openvinotoolkit/open_model_zoo/tree/master/models/public/gpt-2) model, which is a part of the Generative Pre-trained Transformer (GPT) family. GPT-2 is pre-trained on a large corpus of English text using unsupervised training. The model is available from [Open Model Zoo](https://github.com/openvinotoolkit/open_model_zoo/), which we will use to download and convert the model to OpenVINO IR."
```

**Comment by dwipddalal** [comment]
I am guessing the reason for the failure of two test cases is because of some issues with the external packages?

**Comment by adrianboguszewski** [comment]
We don't for now. I'm trying to investigate.

**Comment by dwipddalal** [comment]
It's great to see that all test cases got passed. Now I shall get good sleep.
I tried a lot to find the exact issue but wasn't able to find one.  I would really like to learn the problem that caused the issue, if it's not trouble for you can you please explain it?

**Review by ryanloney — APPROVED** [review]

File: notebooks/223-gpt2-text-prediction/README.md
Comment (adrianboguszewski) [inline notebooks/223-gpt2-text-prediction/README.md:2]:
```suggestion
Text generation is a type of natural language processing that uses computational linguistics and artificial intelligence to automatically produce text that can meet specific communicative needs.
```

**Comment by adrianboguszewski** [comment]
It was a random error of CI.

**Comment by dwipddalal** [comment]
Thanks a lots @adrianboguszewski and @ryanloney  for your help in improving the quality of the notebook. It feels very good to finally get my PR merged into the main branch.

**Pull request merged**
